tests/core/test_cli.py

```diff
@@ -1,45 +1,222 @@
-import unittest.mock
+import unittest
+import unittest.mock as mock
 
 import guarddog.cli
+import guarddog.scanners.scanner as scanner
 from guarddog.ecosystems import ECOSYSTEM
 
 
-def test_is_local_target():
-    assert guarddog.cli.is_local_target("/tmp/foo")
-    assert guarddog.cli.is_local_target("./foo")
-    assert guarddog.cli.is_local_target("../foo")
-    assert guarddog.cli.is_local_target(".")
-    assert not guarddog.cli.is_local_target("foo")
-
-    with unittest.mock.patch("os.path.exists") as mock:
-        mock.return_value = True
-        assert guarddog.cli.is_local_target("foo.tar.gz")
-
-    with unittest.mock.patch("os.path.exists") as mock:
-        mock.return_value = False
-        assert not guarddog.cli.is_local_target("foo.tar.gz")
-
-
-def test_get_rule_param_include():
-    """
-    Test the parsing function returns the included parameter
-    """
-    rules = guarddog.cli._get_rule_param(("shady-links",), (), ECOSYSTEM.NPM)
-    assert rules
-    assert len(rules) == 1
-
-def test_get_rule_param_exclude():
-    """
-    Test the parsing function returns a list without the excluded parameter
-    """
-    rules = guarddog.cli._get_rule_param((), ("shady-links",), ECOSYSTEM.PYPI)
-    assert rules
-    assert len(rules) != 1
-    assert "shady-links" not in rules
-
-def test_get_rule_param_empty():
-    """
-    Test the parsing function returns None when no rules are provided
-    """
-    rules = guarddog.cli._get_rule_param((), (), ECOSYSTEM.PYPI)
-    assert rules is None
+class TestCli(unittest.TestCase):
+
+    def test_local_target(self):
+        # /tmp/foo is a directory
+        with mock.patch("os.path.isdir") as isdir:
```

Comment:
here you could reduce the specific cases by parametrizing tests:
```
@pytest.mark.parametrize("isdir, isfile, expected_list, not_expected_list", 
[True,False,["expected str1",...], ["not expected string 1", "not expected string 2"],
...
[False,True,["expected str1",...], ["not expected string 1", "not expected string 2"],
])
```

Reply:
It turns out that `@pytest.mark.parametrize` is one of the few things in `pytest` that does not work with `unittest`, which is unfortunate.  However, I can still kind of simulate it.